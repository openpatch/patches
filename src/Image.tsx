import { HTMLAttributes } from "react";
import { border, BorderProps, space, SpaceProps } from "./system";

import styled from "@emotion/styled";

export type ImageProps = BorderProps &
  SpaceProps &
  HTMLAttributes<HTMLImageElement>;

// @ts-ignore
export const Image = styled.img<ImageProps>(border, space);
