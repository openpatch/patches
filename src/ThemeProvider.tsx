import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
