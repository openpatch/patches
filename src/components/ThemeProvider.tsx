import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
