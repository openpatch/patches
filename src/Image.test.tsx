import { render } from "@testing-library/react";
import { Image, ImageProps } from "./Image";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: ImageProps) =>
  render(
    <ThemeProvider>
      <Image {...props} />
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
