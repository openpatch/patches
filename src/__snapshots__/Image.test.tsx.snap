// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <img
    class="emotion-0 emotion-1"
  />
</div>
`;
